src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -1735,6 +1717,62 @@ void MarkField (FieldDefinition field, in DependencyInfo reason, in MessageOrigi
 			}
 		}
 
+		bool ShouldWarnForReflectionAccessToCompilerGeneratedCode (FieldDefinition field)
+		{
+			if (!CompilerGeneratedState.IsNestedFunctionOrStateMachineMember (field))
+				return false;
+
+			// No need to warn if it's already covered by the Requires attribute or explicit annotations on the field.
+			if (Annotations.DoesFieldRequireUnreferencedCode (field, out _) || Annotations.FlowAnnotations.ShouldWarnWhenAccessedForReflection (field))
```

Comment:
Should the second case be `!Annotations.FlowAnnotations.ShouldWarnWhenAccessedForReflection`? Or do we skip because it will already warn elsewhere?